src/components/otp-exercise/otp.js

```diff
@@ -1,86 +1,86 @@
-// import {TextInput, View, StyleSheet, StatusBar, Alert} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import {TextInput, View, StyleSheet, StatusBar, Alert} from 'react-native';
 
-// const inputs = new Array(6).fill(0);
+const inputs = new Array(6).fill(0);
 
-// function OTP() {
-//   const [otpTextInput] = useState([]);
-//   const [otp, setOtp] = useState(Array(6).fill(''));
+function OTP() {
+  const [otpTextInput] = useState([]);
+  const [otp, setOtp] = useState(Array(6).fill(''));
 
-//   useEffect(() => {
-//     otpTextInput[0].focus();
-//   }, []);
+  useEffect(() => {
+    otpTextInput[0].focus();
+  }, []);
 
-//   const focusPrevious = (key, index) => {
-//     if (key === 'Backspace' && index !== 0) otpTextInput[index - 1].focus();
-//   };
+  const focusPrevious = (key, index) => {
+    if (key === 'Backspace' && index !== 0) otpTextInput[index - 1].focus();
+  };
 
-//   const focusNext = (index, value) => {
-//     const newOtp = [...otp];
-//     newOtp[index] = value;
+  const focusNext = (index, value) => {
+    const newOtp = [...otp];
+    newOtp[index] = value;
 
-//     if (index < otpTextInput.length - 1 && value) {
-//       otpTextInput[index + 1].focus();
-//     }
+    if (index < otpTextInput.length - 1 && value) {
+      otpTextInput[index + 1].focus();
+    }
 
-//     if (index === otpTextInput.length - 1 && value) {
-//       otpTextInput[index].blur();
-//       newOtp.every(value => value.length && !isNaN(value)) &&
-//         Alert.alert('OTP', `OTP entered is ${newOtp.join('')}`);
-//     }
+    if (index === otpTextInput.length - 1 && value) {
+      otpTextInput[index].blur();
+      newOtp.every(value => value.length && !isNaN(value)) &&
+        Alert.alert('OTP', `OTP entered is ${newOtp.join('')}`);
+    }
 
-//     setOtp(newOtp);
-//   };
+    setOtp(newOtp);
+  };
 
-//   const renderItem = index => {
-//     return (
-//       <View key={index} style={styles.item}>
-//         <TextInput
-//           style={styles.otpField}
-//           keyboardType="numeric"
-//           onChangeText={value => focusNext(index, value)}
-//           onKeyPress={e => focusPrevious(e.nativeEvent.key, index)}
-//           maxLength={1}
-//           onSubmitEditing={() => {
-//             otpTextInput[index].blur();
-//             otp.every(value => value.length && !isNaN(value)) &&
-//               Alert.alert('OTP', `OTP entered is ${otp.join('')}`);
-//           }}
-//           ref={ref => (otpTextInput[index] = ref)}
-//         />
-//       </View>
-//     );
-//   };
+  const renderItem = index => {
+    return (
+      <View key={index} style={styles.item}>
+        <TextInput
+          style={styles.otpField}
+          keyboardType="numeric"
+          onChangeText={value => focusNext(index, value)}
+          onKeyPress={e => focusPrevious(e.nativeEvent.key, index)}
+          maxLength={1}
+          onSubmitEditing={() => {
```

Comment:
This function shouldn't be needed. Can be handled via state change

Reply:
And avoid arrow functions whereever they can be

Reply:
needed if entered on any input field